presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -767,108 +786,256 @@ public static String keyFromPath(Path path)
-                    .withEncryptionMaterials(encryptionMaterialsProvider.get())
-                    .withClientConfiguration(clientConfig)
-                    .withMetricsCollector(metricCollector);
+        // Configure SSL/TLS settings if needed
+        if (!sslEnabled) {
+            log.warn("SSL is disabled - this is not recommended for production use");
         }
-        else {
-            clientBuilder = AmazonS3Client.builder()
-                    .withCredentials(credentialsProvider)
-                    .withClientConfiguration(clientConfig)
-                    .withMetricsCollector(metricCollector);
+
+        String endpoint = hadoopConfig.get(S3_ENDPOINT);
+        boolean isHttpEndpoint = false;
+
+        // Check if endpoint is HTTP (non-secure)
+        if (endpoint != null) {
+            try {
+                URI endpointUri = URI.create(endpoint);
+                if (endpointUri.getScheme() == null) {
+                    endpoint = (sslEnabled ? "https://" : "http://") + endpoint;
+                    endpointUri = URI.create(endpoint);
+                }
+                isHttpEndpoint = "http".equalsIgnoreCase(endpointUri.getScheme());
+                if (isHttpEndpoint) {
+                    log.debug("HTTP endpoint detected: %s - will disable checksum validation", endpoint);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                log.error("Invalid S3 endpoint URL: %s", endpoint);
+                throw new RuntimeException("Invalid S3 endpoint configuration", e);
+            }
         }
 
+        // **FIX: Build S3Configuration with checksum validation disabled for HTTP endpoints**
+        // This prevents the x-amz-content-sha256 mismatch error when using HTTP endpoints
+        // See: https://github.com/aws/aws-sdk-java-v2/issues/5498
+        final boolean disableChecksums = isHttpEndpoint;
+        S3Configuration s3Configuration = S3Configuration.builder()
+                .checksumValidationEnabled(!disableChecksums)
+                .build();
+
+        Optional<Keyring> keyring = createClientSideEncryptionKeyring(hadoopConfig);
+        if (keyring.isPresent()) {
+            log.debug("Creating S3 client with client-side encryption");
+            return createS3EncryptionClient(hadoopConfig, httpClientBuilder.build(),
+                    s3Configuration, keyring.get());
+        }
+
+        S3ClientBuilder clientBuilder = S3Client.builder()
+                .credentialsProvider(credentialsProvider)
+                .httpClient(httpClientBuilder.build())
+                .serviceConfiguration(s3Configuration)
+                .overrideConfiguration(builder -> {
+                    builder.retryStrategy(retryStrategy -> retryStrategy.maxAttempts(maxErrorRetries))
+                            .putAdvancedOption(USER_AGENT_PREFIX,
+                                    userAgentPrefix + " " + S3_USER_AGENT_SUFFIX);
```

Comment:
Let's keep prefix and suffix in separate arguments

```suggestion
                            .putAdvancedOption(USER_AGENT_PREFIX, userAgentPrefix)
                            .putAdvancedOption(USER_AGENT_SUFFIX, S3_USER_AGENT_SUFFIX)
```